core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -683,3 +729,56 @@ trait TemporalSchema {
     )(formatter)
 
 }
+
+case class FieldType(isInput: Boolean, isSubscription: Boolean)
+
+case class PartiallyAppliedField[V](
+  name: String,
+  description: Option[String],
+  directives: List[Directive]
+) {
+  def apply[R, V1](fn: V => V1)(implicit ev: Schema[R, V1], ft: FieldType): (__Field, V => Step[R]) =
+    (
+      __Field(
+        name,
+        description,
+        Nil,
+        () =>
+          if (ev.optional) ev.toType_(ft.isInput, ft.isSubscription)
+          else Types.makeNonNull(ev.toType_(ft.isInput, ft.isSubscription)),
+        directives = Some(directives).filter(_.nonEmpty)
```

Comment:
Should it be `None` if empty? not sure if it makes a difference

Reply:
I don't remember why we used `Option` in some cases and `empty` in others tbh. So I just copied what I had elsewhere.